test/helpers/kubectl.go

```diff
@@ -476,19 +476,23 @@ func (kub *Kubectl) WaitForKubeDNSEntry(name string) error {
 	}
 	// https://bugs.launchpad.net/ubuntu/+source/bind9/+bug/854705
 	digCMD := "dig +short %s @%s | grep -v -e '^$'"
+	digCMDFallback := "dig +tcp %s @%s"
 
-	podsIPs, err := kub.GetPodsIPs(KubeSystemNamespace, "k8s-app=kube-dns")
+	host, _, err := kub.GetServiceHostPort(KubeSystemNamespace, "kube-dns")
 	if err != nil {
-		logger.WithError(err).Error("cannot get kube-dns pods")
+		logger.WithError(err).Error("cannot get kube-dns service IP")
 		return err
 	}
 
 	body := func() bool {
 		var result bool
-		for _, ip := range podsIPs {
-			res := kub.Exec(fmt.Sprintf(digCMD, name, ip))
-			result = res.WasSuccessful()
```

Comment:
Interesting. This loop would try all of the kube-dns pod IPs and result would be the last one in the loop. Is it possible that we should have `if res.WasSuccessful() { break; }` to stop trying when any of them responded correctly? This would handle bad pod IPs. Alternatively, we do `if !res.WasSuccessful() { break; }` and that would test that all of them succeeded (because we return on any fail).

Reply:
We're using the service instead of the DNS, that it's how the pods are using, so I think that it's s better way doing as pods try.